src/main/java/com/google/api/gax/bundling/ThresholdBundler.java

```diff
@@ -154,32 +151,41 @@ public ThresholdBundleHandle add(E e) {
     final Lock lock = this.lock;
     lock.lock();
     try {
+      for (BundlingThreshold<E> threshold : thresholdPrototypes) {
```

Comment:
We could make L154-159 a helper method called "validateItemSize" to make this flow more readable